### ISSUE TITLE
Require reflection for disabling cops

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14740]
## Is your feature request related to a problem? Please describe.

It's arguably "too easy" to just disable a cop. Like a free bribe oxymoron.

## Describe the solution you'd like

For every `Enabled: false`:

1. Require a `Reason:` to make users reflect on their reluctance (and prompt imminent reconsideration)
2. Generate a `Count:` for each full directory tree run that shows magnitude of malpractice
  - Low counts: Low hanging fruit to fix and get rid of cop altogether
  - High counts: Indentifies greater pattern of blind spot in programming habits
3. Sort statements by count ascending

## Describe alternatives you've considered

- Roll own (sort of did).
- (In addition), a wiki that specifically targets the (imagined) hurdles that maintain old habits, shows flawed premises of objections and how it's actually easier and better than feared. A learning journey, rather than a blunt "Bad" / "Good" example. And for the love of OOP, no "Car"/ "Wheel" examples!! 😖

> If One Is Truly to Succeed in Leading a Person to a Specific Place, One Must First and Foremost Take Care to Find Him Where He is and Begin There. – Søren Kierkegaard

## Additional context

Verbose form:

```yaml
Style/GlobalVars:
  Enabled: false
  Reason: Will cause too much boilerplate, everything must be shared/passed around
  Count: 64
```

Terse form (ie. any value that is not a single lowercase word will mean `Enabled: false`)

```yaml
Style/GlobalVars:
  Enabled: (64) Will cause too much boilerplate, everything must be shared/passed around
```

**Comment by lovro-bikic** [comment]
This suggestion reminded me of the [`Bundler/GemComment`](https://docs.rubocop.org/rubocop/cops_bundler.html#bundlergemcomment) cop. It's similar in nature, but the reason needs to be added as a comment above each gem. Perhaps using comments makes sense here as well; I'm not sure if the reason needs to be used programmatically in code, and therefore warrants a YAML property?

Anyway, if there were such a cop, I'd use it on my projects because I think it's important to document the rationale behind disabling a cop.

Regarding the count, there's the autogenerated [`.rubocop_todo.yml`](https://docs.rubocop.org/rubocop/configuration.html#automatically-generated-configuration) file that includes offense count for each disabled cop. Perhaps that helps?

**Comment by forthrin** [comment]
@lovro-bikic: Came up with a good way forward that lifts the issue to a higher level and provides a richer, more complete feature.

Having the code editor show the following in a temporary buffer / mini-window. This yields:

- Side-by-side display provides immediate insight in suggested rewrite
- Reasons the change as to enrich and onboard the developer
- References the manual for full details

```ruby
# Current code
names = users.map { |user| user.name }

# Recommended rewrite
names = users.map(&:name)

# Style/SymbolProc
#
# Prefer `(&:name)` when a block only forwards its argument to one method.
# This removes unnecessary block syntax while preserving behavior and readability.
# The block form remains appropriate when the operation contains additional logic.
#
# https://docs.rubocop.org/rubocop/cops_style.html#stylesymbolproc
```
